dashboard/modules/log/log_manager.py

```diff
@@ -171,8 +198,48 @@ async def resolve_filename(
                     log_filename = filename
                     break
         elif task_id:
-            raise NotImplementedError("task_id is not supported yet.")
+            reply = await self.client.get_task_info(task_id=task_id, timeout=timeout)
+            # Check if the task is found.
+            if len(reply.events_by_task) == 0:
+                raise FileNotFoundError(
+                    f"Could not find log file for task: {task_id}"
+                    f" (attempt {attempt_number}) with suffix: {suffix}"
+                )
+            task_event = None
+            for t in reply.events_by_task:
+                if t.attempt_number == attempt_number:
+                    task_event = t
+                    break
+
+            if task_event is None:
+                raise FileNotFoundError(
+                    "Could not find log file for task attempt:"
+                    f"{task_id}({attempt_number})"
+                )
+
+            # Get the worker id and node id.
+            task = protobuf_to_task_state_dict(task_event)
+
+            worker_id = task.get("worker_id", None)
+            node_id = task.get("node_id", None)
+
+            if worker_id is None or node_id is None:
+                raise FileNotFoundError(
+                    "Could not find log file for task attempt:"
+                    f"{task_id}({attempt_number})."
+                    f"Worker id = {worker_id}, node id = {node_id}"
+                )
+
+            log_filename = await self._resolve_worker_file(
```

Comment:
Hmm don't we already have the API to find this? I assume we should have one given we already support finding the worker log files

Reply:
ah exactly - that's why I refactored it in the first place.